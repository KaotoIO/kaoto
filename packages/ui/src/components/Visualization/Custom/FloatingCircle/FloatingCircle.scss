@use '../custom';

.floating-circle {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--custom-node-BackgroundColor);
  width: 25px;
  height: 25px;
  border-radius: 50%;
  border: 1px solid var(--custom-node-BorderColor);
  box-shadow: var(--custom-node-Shadow);

  [data-selected='true'] & {
    @include custom.selected;
  }
}
