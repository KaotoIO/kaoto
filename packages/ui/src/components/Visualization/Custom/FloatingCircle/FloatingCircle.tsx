import clsx from 'clsx';
import { FunctionComponent, PropsWithChildren } from 'react';
import './FloatingCircle.scss';

export const FloatingCircle: FunctionComponent<PropsWithChildren<{ className?: string }>> = ({
  children,
  className,
}) => {
  return <div className={clsx('floating-circle', className)}>{children}</div>;
};
