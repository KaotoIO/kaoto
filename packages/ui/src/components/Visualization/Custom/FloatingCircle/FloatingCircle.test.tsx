import { render } from '@testing-library/react';
import { FloatingCircle } from './FloatingCircle';

describe('FloatingCircle', () => {
  it('renders children correctly', () => {
    const { container } = render(<FloatingCircle>Test Content</FloatingCircle>);
    expect(container).toMatchSnapshot();
  });

  it('applies custom className', () => {
    const { container } = render(<FloatingCircle className="my-class">Content</FloatingCircle>);
    expect(container).toMatchSnapshot();
  });

  it('renders without children', () => {
    const { container } = render(<FloatingCircle />);
    expect(container).toMatchSnapshot();
  });
});
