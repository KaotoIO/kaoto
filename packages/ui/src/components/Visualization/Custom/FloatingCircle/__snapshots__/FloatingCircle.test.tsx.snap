// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingCircle applies custom className 1`] = `
<div>
  <div
    class="floating-circle my-class"
  >
    Content
  </div>
</div>
`;

exports[`FloatingCircle renders children correctly 1`] = `
<div>
  <div
    class="floating-circle"
  >
    Test Content
  </div>
</div>
`;

exports[`FloatingCircle renders without children 1`] = `
<div>
  <div
    class="floating-circle"
  />
</div>
`;
